src/DynamicData.Tests/Cache/MergeChangeSetsFixture.cs

```diff
@@ -0,0 +1,1056 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Diagnostics.CodeAnalysis;
+using System.Linq;
+using System.Reactive.Linq;
+using DynamicData.Kernel;
+using DynamicData.Tests.Utilities;
+using FluentAssertions;
+using Microsoft.Reactive.Testing;
+using Xunit;
+
+namespace DynamicData.Tests.Cache;
+
+public sealed class MergeChangeSetsFixture : IDisposable
```

Comment:
Very impressive level of testing

Reply:
To be fair, I copied over most of the tests from the `MergeManyChangeSets` fixture and then tweaked them.  Saving all that time allowed me to just add a few more, which is how found one corner case that I had missed in `MergeManyChangeSets`, but now that the implementation is shared, fixing one fixed them both.

But it was a real corner-corner case that required using a custom EqualityComparer and custom Comparer together.

The issue was:
1) Values are added (merged results sees the add)
2) Values are updated to values that are equal according to the EqualityComparer (merge results sees nothing)
3) Values are refreshed (merged results sees refresh events for objects it has never observed) and they still are equal to the originally added (and updated) values

I changed it so that now, at 3, it won't see the refresh events either because something shouldn't be refreshed if it hasn't been seen.

I'm pretty sure that's the right behavior.  If the refresh had caused the updated value to be the new best value, then it would have observed the Update at that point.  Tell me if you think that's the wrong behavior.